### PR TITLE
Setup readinglist documentation. (refs #18, refs #19)

### PR DESCRIPTION
**Pull request by Natim**
Is it enough for you or do you want to spawn a virtualenv for the documentation dependencies?

### CODE REVIEW

**Comment by Natim** [comment]
R? @leplatrem

**Comment by leplatrem** [comment]
What do you think if we only put what's already implemented ?

When we'll implement a new feature, we'll copy-paste the piece of doc from the Wiki and put it here in the docs.

This way, there is no confusion between *planned* and *available* features. Plus, we can also make sure the implementation matches the documentation during pull request review

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.. Reading List documentation master file, created by
+   sphinx-quickstart on Mon Feb  2 15:08:06 2015.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to Reading List's documentation!
```
Comment (leplatrem) [inline docs/index.rst]: Reading List documentation

Usually you would use the apostrophe when it means "possession" (like Remy's flat)

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+========================================
+
+*Reading list* is a service to keep track of a list of articles to read.
+The state of the reading list is synchronized between devices.
```
Comment (leplatrem) [inline docs/index.rst]: This is a service attached to Firefox Accounts that enable users to keep their Reading List in sync across all of their devices.

source: https://wiki.mozilla.org/Mobile/FxA_-_Reading_List

File: docs/index.rst (hunk 1 above)
Comment (Natim) [inline docs/index.rst]: Well this is made up by Sphinx quickstart :)

Comment (Natim) [inline docs/index.rst]: Plus it looks right according to: http://english.stackexchange.com/a/6736

File: docs/Makefile [hunk 3]
```diff
@@ -0,0 +1,180 @@
+# Makefile for Sphinx documentation
```
Comment (ametaireau) [inline docs/Makefile]: I always feel this Makefile is kinda superfluous: we probably should just do the call to sphinx-build and that would be easier: we don't use the other targets.

Comment (Natim) [inline docs/Makefile]: I like to keep the Sphinx default Makefile for all projects. It helps people coming to the project to feel home.

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+GET /
+-----
+
+The returned value is a JSON mapping containing:
+
+- ``hello``: the name of the service (``"reading list"``)
+- ``version``: complete version (``"X.Y.Z"``)
+- ``url``: absolute URI (without a trailing slash) of the API (*can be used by client to build URIs*)
+- ``eos``: date of end of support in ISO 8601 format (``"yyyy-mm-dd"``, null if unknown)
+- ``documentation``: The url to the service documentation
```
Comment (ametaireau) [inline docs/api.rst]: we should add "(here !)"

File: docs/api.rst [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+
+
+GET /__heartbeat__
+------------------
+
+Return the status of each service the *reading list* depends on. The returned value is a JSON mapping containing:
+
+- ``database`` true if operational
+
+Return ``200`` if the connection with each service is working properly and ``503`` if something doesn't work.
```
Comment (ametaireau) [inline docs/api.rst]: Hmm, we should wrap at 80 chars :)

File: docs/api.rst [hunk 6]
```diff
@@ -0,0 +1,537 @@
+#################
+API Documentation
+#################
+
+
+GET /
+-----
```
Comment (ametaireau) [inline docs/api.rst]: We probably should use `====` for first level titles, and `----` for second level ones.

File: docs/index.rst (hunk 1 above)
Comment (ametaireau) [inline docs/index.rst]: Anyway, "welcome to" seems weird to me. I believe we should remove the "welcome" part.

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+#texinfo_appendices = []
+
+# If false, no module index is generated.
+#texinfo_domain_indices = True
+
+# How to display URL addresses: 'footnote', 'no', or 'inline'.
+#texinfo_show_urls = 'footnote'
+
+# If true, do not generate a @detailmenu in the "Top" node's menu.
+#texinfo_no_detailmenu = False
```
Comment (ametaireau) [inline docs/conf.py]: Wow, this is big.

File: docs/backoff.rst [hunk 8]
```diff
@@ -0,0 +1,36 @@
+##################
+Backoff indicators
+##################
+
+Backoff header on heavy load
```
Comment (ametaireau) [inline docs/backoff.rst]: We probably should add anchors here so that we can link there from other docs easily.

**Comment by Natim** [comment]
Updated.

File: docs/api.rst [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+==================
+
+Return the status of each service the *reading list* depends on. The
+returned value is a JSON mapping containing:
+
+- ``database`` true if operational
+
+Return ``200`` if the connection with each service is working properly
+and ``503`` if something doesn't work.
+
```
Comment (leplatrem) [inline docs/api.rst]: What do you think if we put these two *generic* endpoints in a separate doc file ?

File: docs/authentication.rst [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+**Reading list scope**
+
+The *reading list* API will eventually have to handle a dedicated OAuth scope (e.g.
+``readinglist``, ``readinglist:read``, ``readinglist:write``). This will help users
+to delegate access to the readinglist to third party apps
+
+So far the FxA server only handles the ``profile`` scope.
+
+See https://github.com/mozilla-services/readinglist/issues/16.
```
Comment (leplatrem) [inline docs/authentication.rst]: We could mention the BasicAuth configuration for local testing or during development

File: docs/api.rst [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+If the request header ``If-Unmodified-Since`` is provided, and if the record has
+changed meanwhile, a ``412 Precondition failed`` error is returned.
+
+:note:
+    Once deleted, an article will appear in the collection with a deleted status
+    (``status=2``) and will have most of its fields empty.
+
+:note:
+    The server will have to implement an internal mechanism to will keep track of deleted items,
+    and purge them eventually.
```
Comment (leplatrem) [inline docs/api.rst]: Since this is not implemented, should we mention it here ?

File: docs/api.rst [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+    Changing ``read_position`` never generates conflicts.
+
+:note:
+    ``read_position`` can only be changed for a greater value than the current one.
+
+:note:
+    If ``unread`` is changed to false, ``marked_read_on`` and ``marked_read_by`` are expected to be provided.
+
+:note:
+    If ``unread`` was already false, ``marked_read_on`` and ``marked_read_by`` are not updated with provided values.
```
Comment (leplatrem) [inline docs/api.rst]: Since these 3 details are not implemented, should we mention them here ?

File: docs/api.rst [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+
+:note:
+    Deleted items should be taken into account for URL unicity.
+
+If an article is created with an URL that already exists, a ``303 See Other`` response
+is returned to indicate the existing record.
+
+The response body is a JSON mapping, with the following attribute:
+
+- ``id``: the id of the conflicting record
```
Comment (leplatrem) [inline docs/api.rst]: Not implemented either

File: docs/api.rst [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+If the posted values are invalid (e.g. *added_on is not an integer*) an error response is returned with status ``400``. `See details on error responses <https://github.com/mozilla-services/readinglist/wiki/API-Design-proposal#error-responses>`_.
+
+
+:note:
+    The ``status`` can take only ``0`` (OK) and ``1`` (archived), even though
+    the server sets it to ``2`` when including deleted articles in the collection.
+
+:note:
+    *(undecided)* For some cases, it can make sense for the server to fix arbitrarily
+    validation errors on records (e.g. truncating long titles).
```
Comment (leplatrem) [inline docs/api.rst]: not decided

File: docs/api.rst [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+
+- ``<prefix?><attribute name>``: filter by value(s)
+- ``_since``: polling changes
+- ``_sort``: order list
+- ``_limit``: pagination max size
+
+Some additional internal parameters are used by pagination. Client should not
+be aware of them, since they are set and provided through the ``Next-Page`` header.
+
+- ``_page_token``: pagination continuation token
```
Comment (leplatrem) [inline docs/api.rst]: Not decided / implemented

**Comment by Natim** [comment]
Updated.

**Pull request merged**